optimum/neuron/trainers.py

```diff
@@ -402,20 +404,27 @@ def _save_xla(self, output_dir: Optional[str] = None):
             xm.mark_step()
             parallelizer.save_model_checkpoint(self.model, output_dir, as_sharded=True, optimizer=self.optimizer)
         else:
+            safe_save_function_patcher = Patcher(
+                [("transformers.modeling_utils.safe_save_file", torch_xla_safe_save_file)]
+            )
             if not isinstance(self.model, PreTrainedModel):
                 if isinstance(unwrap_model(self.model), PreTrainedModel):
-                    unwrap_model(self.model).save_pretrained(
-                        output_dir,
-                        is_main_process=self.args.should_save,
-                        state_dict=self.model.state_dict(),
-                        save_function=xm.save,
-                    )
+                    with safe_save_function_patcher:
+                        unwrap_model(self.model).save_pretrained(
```

Comment:
1. Could you explain what you want to avoid/achieve here ?
2. Could you explain what will the exact chain of calls be here ?

Reply:
So basically the chain of calls here is:

1. We enter a context manager that patches the `transformers.modeling_utils.safe_save_file` function which is `safetensors.torch.save_file` to be my `torch_xla` compatible version of this function.
2. The model is unwraped (it's not really important here, it's related to the Trainer and how sometimes the model is wrapped for feature I am not really sure we support).
3. The model is saved, and since last Transformers release, it will save the checkpoint using `safetensors`. [The issue is that when `safetensors` is used, the `save_function` parameter is ignored](https://github.com/huggingface/transformers/blob/main/src/transformers/modeling_utils.py#L2372-L2378). *That is the reason why we do it like that instead of simply passing the `torch_xla_safe_save_file` as the value for `save_function`.
4. We exit the context manager and everything that was patched is restored to its original value.

Reply:
Thanks for the explaination !